lib/Disco.php

```diff
@@ -844,4 +874,43 @@ private static function constructSearchData($idpMetadata): string
 
         return strtolower(str_replace('"', '', iconv('UTF-8', 'US-ASCII//TRANSLIT', $res)));
     }
+
+    private static function substrInArray($needle, array $haystack)
+    {
+        foreach ($haystack as $item) {
+            if (strpos($item, $needle) !== false) {
+                return $item;
+            }
+        }
+
+        return null;
+    }
+
+    private function fillSpName($state)
+    {
+        $translate = new Translate(Configuration::getInstance());
+
+        $clientIdWithPrefix = self::substrInArray(self::CLIENT_ID_PREFIX, $this->originalAuthnContextClassRef);
+
+        if ($clientIdWithPrefix !== null) {
+            $parts = explode(':', $clientIdWithPrefix);
+            $clientId = end($parts);
+
+            $this->adapter = Adapter::getInstance($this->wayfConfiguration->getString(self::INTERFACE, self::RPC));
+
+            $facility = $this->adapter->getFacilityByClientId($clientId);
```

Comment:
Just to be sure, I would probably catch any exceptions thrown by the method. If we do not have the name, disco should work anyway.